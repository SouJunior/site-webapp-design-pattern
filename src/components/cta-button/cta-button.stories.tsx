import type { Meta, StoryObj } from "@storybook/react";
import { CtaButton } from "./cta-button";

const meta: Meta<typeof CtaButton> = {
  title: "Components/CtaButton",
  component: CtaButton,
  tags: ["autodocs"],
  argTypes: {
    label: { control: "text" },
    onClick: { action: "clicked" },
  },
};

export default meta;
type Story = StoryObj<typeof CtaButton>;

export const Default: Story = {
  args: {
    label: "FaÃ§a Parte",
  },
};

export const WithEmoji: Story = {
  args: {
    label: "ðŸš€ Clique Aqui",
  },
};
