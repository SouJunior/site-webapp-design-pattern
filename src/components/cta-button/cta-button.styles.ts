import styled from "styled-components";

export const StyledButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  gap: 0.5rem;

  padding: 0.75rem 1.5rem;
  background-color: #0c2f87;
  color: white;
  font-size: 1rem;
  font-weight: bold;
  border: none;
  border-radius: 999px;

  cursor: pointer;
  transition: background-color 0.2s ease;

  &:hover {
    background-color: #09256b;
  }
`;

export const ArrowIcon = styled.img`
  width: 1rem;
  height: 1rem;
`;
