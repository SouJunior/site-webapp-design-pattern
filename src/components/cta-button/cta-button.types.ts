import { ButtonHTMLAttributes } from "react";

export interface CtaButtonProps
  extends ButtonHTMLAttributes<HTMLButtonElement> {
  label: string;
}
