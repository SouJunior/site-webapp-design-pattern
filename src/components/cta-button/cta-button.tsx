import { StyledButton, ArrowIcon } from "./cta-button.styles";
import type { CtaButtonProps } from "./cta-button.types";
import arrow1 from "../../assets/arrow1.png";

export const CtaButton = ({ label, ...props }: CtaButtonProps) => (
  <StyledButton {...props} aria-label={label}>
    {label}
    <ArrowIcon src={arrow1} alt="" aria-hidden="true" />
  </StyledButton>
);
