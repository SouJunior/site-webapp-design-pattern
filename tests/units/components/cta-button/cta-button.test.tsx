import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import { CtaButton } from "../../../../src/components/cta-button/cta-button";

describe("CtaButton", () => {
  it("renders with provided label", () => {
    render(<CtaButton label="Clique aqui" />);
    expect(
      screen.getByRole("button", { name: /clique aqui/i })
    ).toBeInTheDocument();
  });

  it("calls onClick handler when clicked", () => {
    const onClick = jest.fn();
    render(<CtaButton label="Clique" onClick={onClick} />);

    fireEvent.click(screen.getByRole("button", { name: /clique/i }));
    expect(onClick).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    const { container } = render(<CtaButton label="Snapshot Test" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
