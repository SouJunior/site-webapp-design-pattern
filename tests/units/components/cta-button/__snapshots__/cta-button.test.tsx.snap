// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CtaButton matches snapshot 1`] = `
<button
  aria-label="Snapshot Test"
  class="sc-VHjGu euCHIR"
>
  Snapshot Test
  <img
    alt=""
    aria-hidden="true"
    class="sc-gnOvAp eYXTkY"
    src="mocked-file"
  />
</button>
`;
